databend-jdbc/src/main/java/com/databend/jdbc/DatabendConnection.java

```diff
@@ -610,47 +617,69 @@ public void accept(DatabendSession session) {
         setSession(session);
     }
 
+    /**
+     * Retry executing a query in case of connection errors. fail over mechanism is used to retry the query when connect error occur
+     * It will find next target host based on configured Load balancing Policy.
+     * @param sql The SQL statement to execute.
+     * @param attach The stage attachment to use for the query.
+     * @return A DatabendClient instance representing the successful query execution.
+     * @throws SQLException If the query fails after retrying the specified number of times.
+     * @see DatabendClientLoadBalancingPolicy
+     */
+    DatabendClient startQueryWithFailover(String sql, StageAttachment attach) throws SQLException {
+        Exception e = null;
+        int times = getMaxFailoverRetries() + 1;
+
+        for( int i = 1; i <= times; i++) {
+            if (e != null && !(e.getCause() instanceof ConnectException)) {
+                throw new SQLException("Error start query: " + "SQL: " + sql + " " + e.getMessage() + " cause: " + e.getCause(), e);
+            }
+            try {
+                if (!inActiveTransaction()) {
+                    this.routeHint = randRouteHint();
+                }
+                // configure query and choose host based on load balancing policy.
+                ClientSettings.Builder sb = this.makeClientSettings();
+                if (attach != null) {
+                    sb.setStageAttachment(attach);
+                }
+                ClientSettings s = sb.build();
+                logger.log(Level.FINE, "retry " + times + " times to execute query: " + sql + " on " + s.getHost());
+                return new DatabendClientV1(httpClient, sql, s, this);
+            } catch (RuntimeException e1) {
```

Comment:
There is no need to catch `RuntimeException` because `RuntimeException` is the child class of `Exception`. Just 
```
catch (Exception e) {
    throw new SQLException(String.format("Error executing query: SQL: %s %s cause: %s", sql, e.getMessage(), e.getCause()), e);
}
```